google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/reframing/ReframingResponseObserver.java

```diff
@@ -193,10 +193,23 @@ private void onCancel() {
    */
   @Override
   protected void onResponseImpl(InnerT response) {
+    boolean shoudCancelStream = false;
+
     synchronized (lock) {
       Preconditions.checkState(awaitingInner, "Received unsolicited response from upstream");
       awaitingInner = false;
-      reframer.push(response);
+
+      try {
+        reframer.push(response);
+      } catch (Throwable t) {
+        if (error != null) {
```

Comment:
Shouldn't this be `error == null`?

Reply:
Fixed, that was stupid mistake. I'm working on a new version of the reframing logic in a different branch and screwed up copying the changes over. Please merge when ready